src/main.zig

```diff
@@ -1711,47 +1712,55 @@ fn buildOutputType(
             if (runtime_args_start) |i| {
                 try argv.appendSlice(all_args[i..]);
             }
-            // TODO On operating systems that support it, do an execve here rather than child process,
-            // when watch=false and arg_mode == .run
-            const child = try std.ChildProcess.init(argv.items, gpa);
-            defer child.deinit();
+            if (std.builtin.os.tag == .linux and arg_mode == .run and watch == false) { // TODO does other posix os support execve in zig stdlib?
+                var env_vars = try process.getEnvMap(gpa);
+                defer env_vars.deinit();
+                const term = os.execvpe(gpa, argv.items, &env_vars);
+                switch (arg_mode) {
```

Comment:
You know for sure that `arg_mode == .run`, the whole switch can be collapsed into `process.exit(1)`.